### DISCUSSION TITLE
Could `crane` run `cargo test --release` on every dependency?

### DISCUSSION THREAD

**Discussion by dpc** [ipetkov/crane#103] [Ideas]
Something that cargo can't do (AFAIK) is running tests on all the dependencies (https://github.com/rust-lang/cargo/issues/483). In certain high stake setups that would be something really valuable to do, just to make sure everything is fine using the exact locked versions.

I was wondering how hard would it be for crane to do it, since it already has the exact list and all the source code. If crane would allow it, I would run it in a once a day CI run.

**Comment by ipetkov** [comment]
It sounds like using `cargo test -p CRATE_NAME` can be used to run the tests for the dependency, but doesn't sound like cargo supports running tests for all deps with a single flag...

Might be possible to script something by pulling the names of all dependencies via `cargo metadata` and invoke `cargo test -p ...` for each one

**Reply by dpc** [reply]
`cargo test -p CRATE` doesn't work if the CRATE needs any dev-dependencies that are not in the current workspace.

**Reply by ipetkov** [reply]
Hmm yeah not sure if this can be done transitively. The `Cargo.lock` file doesn't include dev-deps either, so sounds like the correct way to go about testing dependency crates is to checkout and build their actual workspace

What is the original motivation trying to accomplish? Running the tests for a dependency sounds good and all, but if the goal is to track known/trusted versions of the dependency then running its tests might not be enough

**Reply by dpc** [reply]
We were talking about `-Z minimim-versions` aproaches, and it occurred to me that downstream programs probably care most if the libraries they depend on work OK not with minimum versions, but with the ones they actually select. 

After thinking more about it, I don't think it's very practical, and not not crane's job to do it. If anything, it should be `cargo`'s feature.
